source/common/http/filter_manager.h

```diff
@@ -577,18 +639,11 @@ class FilterManager : public ScopeTrackedObject,
   void callHighWatermarkCallbacks();
   void callLowWatermarkCallbacks();
 
-  void setRequestHeaders(RequestHeaderMapPtr&& request_headers) {
-    if (Http::Headers::get().MethodValues.Head == request_headers->getMethodValue()) {
+  void requestHeadersInitialized() {
```

Comment:
optional nit (and fine to do in a follow up) I think the naming is a bit strange, since requestHeadersInitialized() is used to initialize request header values.  onRequestHeaders() seems more Envoyish.